lib/jormungandr/src/Cardano/Wallet/Jormungandr/Transaction.hs

```diff
@@ -109,13 +117,40 @@ newTransactionLayer (Hash block0) = TransactionLayer
             $ Left ErrExceededInpsOrOuts
     }
 
+-- | Provide a transaction witness for a given private key. The type of witness
+-- if different between types of keys and, with backward-compatible support, we
+-- may have to support many types for one backend target.
+class MkTxWitness (k :: Depth -> * -> *) where
+    mkTxWitness
+        :: Hash "Genesis"
+        -> [(TxIn, Coin)]
+        -> [TxOut]
+        -> (k 'AddressK XPrv , Passphrase "encryption")
+        -> TxWitness
+
+instance MkTxWitness SeqKey where
+    mkTxWitness (Hash block0) inps outs xprv =
+        let
+            payload = block0 <> getHash (signData inps outs)
+        in
+            utxoWitness (sign payload xprv)
+
+instance MkTxWitness RndKey where
+    mkTxWitness (Hash block0) inps outs xprv =
+        let
+            xpub = getRawKey $ publicKey $ fst xprv
+            payload = block0 <> getHash (signData inps outs)
```

Comment:
Nice! 

_Maybe_ a bit unfortunate that the `payload`-calculation is duplicated. Separate "signing" and "payload", or does things stop making sense then?

Reply:
Now it would. Initially even the signing was different and I refactored a few things until the point where it got pretty similar. I ended up with this to also make testing a bit easier and the function signature slightly more readable than having a `ByteString` signature as argument.

Reply:
Done.